Framework/API/src/IPeakFunction.cpp

```diff
@@ -78,21 +80,28 @@ class TempJacobian : public Jacobian {
   std::vector<double> m_J;
 };
 
-/// Default value for the peak radius
-int IPeakFunction::s_peakRadius = 5;
+/// Tolerance for determining the smallest significant value on the peak
+const double PEAK_TOLERANCE = 1e-14;
+
+} // namespace
+
+/// "Infinite" value for the peak radius
+const int IPeakFunction::g_maxPeakRadius = std::numeric_limits<int>::max();
 
 /**
-  * Constructor. Sets peak radius to the value of curvefitting.peakRadius
- * property
+  * Constructor.
+  * @param peakRadius :: The peak radius for this instance.
   */
-IPeakFunction::IPeakFunction() {
-  int peakRadius;
-  if (Kernel::ConfigService::Instance().getValue("curvefitting.peakRadius",
-                                                 peakRadius)) {
-    if (peakRadius != s_peakRadius) {
-      setPeakRadius(peakRadius);
-    }
+IPeakFunction::IPeakFunction(int peakRadius) : m_peakRadius(peakRadius) {}
+
+void IPeakFunction::function(const FunctionDomain &domain,
+                             FunctionValues &values) const {
+  auto peakRadius =
```

Comment:
Could this cause confusion though if a user is trying to set it from `Fit` and then in some circumstances their value is not used?